Add search engine comparison table

redoomed1 opened the pull request:
Changes proposed in this PR:

- Add comparison table for the Search Engines page, following the lead of https://github.com/privacyguides/privacyguides.org/pull/2478
  - Make changes enumerated [here](https://github.com/privacyguides/privacyguides.org/pull/2505#issuecomment-2047823097) and [here](https://github.com/privacyguides/privacyguides.org/pull/2505#issuecomment-2048081218)

<del>As marked on GitHub, this is a draft PR as wording and formatting is not set in stone. I'm unsure of what to rearrange, add, or remove, so any feedback is welcome. Here are some questions as a starting point:
    - **Logging**: Is it better to list information that is logged in the table, and list what isn't logged in the description under each card; or, as in the [DNS comparison table](https://www.privacyguides.org/en/dns/#recommended-providers), just link to a footnote explaining the logging policies?
    - **Countries of operation**: List it in a column, or include it in the descriptions of each recommendation? (The latter is done for the VPN recommendations.)
    - **Metasearch engines**: Being that SearXNG is not like the other providers on this page, does it necessitate a new "Metasearch engines" heading; or should it still be included in the table?</del>

<!-- SCROLL TO BOTTOM TO AGREE!:
Please use a descriptive title for your PR, it will be included in our changelog!

If you are making changes that you have a conflict of interest with, please
disclose this as well (this does not disqualify your PR by any means):

Conflict of interest contributions involve contributing about yourself,
family, friends, clients, employers, or your financial and other relationships.
Any external relationship can trigger a conflict of interest.
-->

<!-- Place an x in the boxes below, like: [x] -->
- [x] I have disclosed any relevant conflicts of interest in my post.
- [x] I agree to grant Privacy Guides a perpetual, worldwide, non-exclusive, transferable, royalty-free, irrevocable license with the right to sublicense such rights through multiple tiers of sublicensees, to reproduce, modify, display, perform, relicense, and distribute my contribution as part of this project.
- [x] I am the sole author of this work. <!-- Do not check this box if you are not -->
- [x] I agree to the [Community Code of Conduct](https://www.privacyguides.org/coc).

<!-- What's this? When you submit a PR, you keep the Copyright for the work you
are contributing. We need you to agree to the above terms in order for us to
publish this contribution to our website. -->

jonaharagon commented on docs/search-engines.md:
```diff
@@ -53,55 +63,59 @@ DuckDuckGo is the default search engine for the Tor Browser and is one of the fe
+Startpage previously placed limitations on VPN and [Tor](tor.md) users, but they recently created an [official](https://support.startpage.com/hc/en-us/articles/24786602537364-Startpage-s-Tor-onion-service) Tor hidden service, and as of April 2024 we have no longer noticed extra roadblocks for Tor or [VPN](vpn.md) users.
 
-## Startpage
+## Metasearch Engines
+
+Metasearch engines allow you to aggregate the results of other search engines, such as the ones recommended above, while not storing any information itself.
```
> Metasearch engines: Being that SearXNG is not like the other providers on this page, does it necessitate a new "Metasearch engines" heading; or should it still be included in the table?

Hmm. I suppose this depends on whether we are recommending someone install SearXNG themselves and use it, or find a public SearXNG provider to use. I thought that we were primarily recommending **not** self-hosting because you'd want your queries to blend in with other instance users, thus it shouldn't be separated and it should be included in the table. Definitely open to this alternative though too.

cc @dngray

dngray replied:
I think this really should be in it's own table. The other options Brave, DDG, SP are not self hostable.



> I thought that we were primarily recommending **not** self-hosting because you'd want your queries to blend in with other instance users

There are pros/cons to both. Public instances are likely to be blocked. iirc that used to be a major problem. I think it was more the warning that if you don't use a public instance or this through VPN then other people's search queries might be tied to you.

redoomed1 replied:
I think that, ideally, the SearXNG recommendation should be housed in the [prospective Self-hosting overview](https://github.com/privacyguides/privacyguides.org/issues/1686) page for somewhat the same reason you outlined in [one of your recent forum comments](https://discuss.privacyguides.net/t/amnezia/17794/7):

> Not the main VPN page, because that has specific criteria on auditing, which is likely to never happen or be applicable to self hosted applications.
---
Until then, I think that creating another category for SearXNG on this page (which I did in the initial draft) doesn't make too much sense upon further reflection.

Yet, I don't think that including SearXNG in the table with the other providers offers much value, if "depends on instance operator/owner" is the answer under every column in the table.

I'll defer to whatever decision y'all decide on :smile:

dngray replied:
Someone should get started on a self hosted page :wink:, and we should move that there.

jonaharagon replied:
Well for the purposes of this PR we can keep it as it is then, and in the future we can move it separately 👍

jonaharagon commented on docs/search-engines.md:
```diff
@@ -4,14 +4,24 @@ title: "Search Engines"
-## Brave Search
+| Provider | Privacy Policy | Search Index | Tor Hidden Service | Logging | Country of Operation
+|---|---|---|---|---|---
+| [Brave Search](#brave-search) | [:octicons-link-external-24:](https://search.brave.com/help/privacy-policy) | Independent | :material-check:{ .pg-green } | OS   Browser in use | United States
+| [DuckDuckGo](#duckduckgo) | [:octicons-link-external-24:](https://duckduckgo.com/privacy) | Bing | :material-check:{ .pg-green } | Search queries | United States
+| [Startpage](#startpage) | [:octicons-link-external-24:](https://startpage.com/en/privacy-policy) | Google and Bing | :material-check:{ .pg-green } | OS   Type of browser   Language | Netherlands
```
```suggestion
| [Brave Search](#brave-search) | [:octicons-link-external-24:](https://search.brave.com/help/privacy-policy) | Independent | :material-check:{ .pg-green } | OS, user agent | United States
| [DuckDuckGo](#duckduckgo) | [:octicons-link-external-24:](https://duckduckgo.com/privacy) | Bing | :material-check:{ .pg-green } | Search queries | United States
| [Startpage](#startpage) | [:octicons-link-external-24:](https://startpage.com/en/privacy-policy) | Google and Bing | :material-check:{ .pg-green } | OS, user agent, language | Netherlands
```

redoomed1 commented:
Changes since the initial commit:

- Removed "Country of Operation" column from the table and incorporated the information into the descriptions of each recommendation
- United the "Logging" and "Privacy Policy" columns to one (idea from https://github.com/privacyguides/privacyguides.org/pull/2509) and added footnotes to each entry in this newly consolidated column

redoomed1 commented:
Changes made before submitting for review:

- Re-added "Country of Operations" column to the table since it looked sparse with just four columns
- Abbreviated "personally identifying information" in accordance with the [style guide](https://www.privacyguides.org/en/meta/writing-style/?h=abbrevi#choose-your-words-carefully)
- Added internal links for Tor Browser and Safari in the DuckDuckGo card
- Added an embedded link to [Wikipedia page about metasearch engines](https://en.wikipedia.org/wiki/Metasearch_engine)
- Added an embedded link to a page explaining Brave Search's Discussions feature
- Changed wording in several places
